doubleChainSwap.md

```diff
@@ -0,0 +1,193 @@
+                        script = sndTokensContract)
+
+// Create the fst order transaction which locks the hodlers funds under the contract
+val fstOrderTransaction = Transaction(
+      inputs        = fstHodler.selectUnspentBoxes(toSpend = fstHodlerFunds),
+      outputs       = List(fstOrderBox),
+      fee           = MinTxFee,
+      sendChangeTo  = fstHodler.wallet.getAddress
+  )
+
+// Print fstOrderTransaction
+println(fstOrderTransaction)
+
+// Sign the fstOrderTransaction
+val fstOrderTransactionSigned = fstHodler.wallet.sign(fstOrderTransaction)
+
+
+// Submit the tx to the simulated blockchain
+blockchainSim.send(fstOrderTransactionSigned)
+fstHodler.printUnspentAssets()
+println("-----------")
+
+
+sndHodler.generateUnspentBoxes(
+      toSpend       = sndHodlerFunds, 
+      tokensToSpend = List(sndToken -> sndTokenAmount))
+sndHodler.printUnspentAssets()
+
+// Create a snd order box with sndHodler's funds locked under the contract
+val sndOrderBox   = Box(value = 1 * nanoergsInErg,
+                        token = (sndToken -> sndTokenAmount),
+                        script = fstTokensContract)
+
+// Create the fst order transaction which locks the hodlers funds under the contract
+val sndOrderTransaction = Transaction(
+      inputs        = sndHodler.selectUnspentBoxes(toSpend = sndHodlerFunds),
+      outputs       = List(fstOrderBox),
+      fee           = MinTxFee,
+      sendChangeTo  = sndHodler.wallet.getAddress
+  )
+
+// Print fstOrderTransaction
+println(sndOrderTransaction)
+
+// Sign the fstOrderTransaction
+val sndOrderTransactionSigned = sndHodler.wallet.sign(sndOrderTransaction)
+
+
+// Submit the tx to the simulated blockchain
+blockchainSim.send(sndOrderTransactionSigned)
+sndHodler.printUnspentAssets()
+println("-----------")
+
+
+///////////////////////////////////////////////////////////////////////////////////
+// Matching Orders //
+///////////////////////////////////////////////////////////////////////////////////
+println("===== Generating Order Matching Tx =====")
+
+
+// Create the box which goes back to fstHodler which holds the SND Tokens
+val fstOut = Box(value  = nanoergsInErg / 2,
+                token = (sndToken -> sndTokenAmount),
+                register = (R4 -> fstOrderTransactionSigned.outputs(0).id),
+                script = contract(fstHodler.wallet.getAddress.pubKey))
+
+// Create the box which goes back to sndHodler which holds the FST Tokens
+val sndOut = Box(value = nanoergsInErg / 2,
+                token = (fstToken -> fstTokenAmount),
+                register = (R4 -> sndOrderTransactionSigned.outputs(0).id),
+                script = contract(sndHodler.wallet.getAddress.pubKey))
+
+// Create the swap transaction
+val swapTransaction = Transaction(
+    inputs = List(sndOrderTransactionSigned.outputs(0), fstOrderTransactionSigned.outputs(0)),
+    outputs = List(fstOut, sndOut),
+    sendChangeTo = fstHodler.wallet.getAddress,
```

Comment:
In this situation, the change is sent to fstHodler. What would instead be the recommended situation? Should the change get burned? Can the change be split and sent to both parties?

Reply:
Different options are possible I guess, you can't burn ERG though. sending to one party is okay I suppose (especially in an example)

Reply:
I reduced the transaction fee to minimum and it solved the issue.